model/project_ref.go

```diff
@@ -2245,54 +2250,78 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 }
 
 // ValidateProjectCreation returns a boolean if you should surface the error or not.
-func ValidateProjectCreation(projectId string, config *evergreen.Settings, projectRef *ProjectRef) (bool, error) {
+func ValidateProjectCreation(projectId string, config *evergreen.Settings, projectRef *ProjectRef) (int, error) {
 	if config.ProjectCreation.TotalProjectLimit == 0 || config.ProjectCreation.RepoProjectLimit == 0 {
-		return false, nil
+		return 0, nil
 	}
-	catcher := grip.NewBasicCatcher()
-	allEnabledProjects, err := GetEnabledProjects()
+	originalRef, err := findOneProjectRefQ(byId(projectId))
 	if err != nil {
-		return true, errors.Wrap(err, "getting number of projects")
+		return http.StatusInternalServerError, errors.Wrapf(err, "getting project '%s'", projectId)
 	}
-	allEnabledProjectsId := []string{}
-	for _, p := range allEnabledProjects {
-		allEnabledProjectsId = append(allEnabledProjectsId, p.Id)
+	if originalRef != nil {
+		if projectRef == nil {
+			// Should be unreachable
+			return http.StatusInternalServerError, errors.New("cannot create project while defaulting to repo")
+		}
+		originalRef, err = GetProjectRefMergedWithRepo(*originalRef)
+		if err != nil {
+			return http.StatusInternalServerError, errors.Wrapf(err, "getting merged project for '%s'", originalRef.Id)
+		}
 	}
-	// No need to validate if project was enabled to begin with.
-	if utility.StringSliceContains(allEnabledProjectsId, projectId) {
-		return false, nil
+	var newRef *ProjectRef
+	if projectRef == nil {
+		// Default to repo case
+		repoRef, err := FindOneRepoRef(originalRef.Id)
```

Comment:
Should add a comment to explain why this needs repo ref.

Reply:
deleted